### ISSUE TITLE
cli: incorrect top-level usage output when subcommand provides invalid optional parameter 

### ISSUE THREAD

**Issue by blackboxsw** [canonical/cloud-init#4609]
# Bug report
cloud-init provides invalid usage statements when `--invalid` optional parameters are provided to a cloud-init sub-commands.

The usage reported on the command line is the usage of the base command `cloud-init` when it should be the usage of the specific usage of the sub-command.

## Expected:
Providing and invalid optional parameter `--invalid` to a cloud-init sub-command should generate a usage statement for that sub-command
```

$ cloud-init clean --invalid

usage: /usr/bin/cloud-init clean [-h] [-l] [--machine-id] [-r] [-s] [-c {all,ssh_config,network} [{all,ssh_config,network} ...]]
/usr/bin/cloud-init clean: error: unrecognized arguments: --invalid
```

## Actual:

Providing and invalid optional parameter --invalid to a cloud-init sub-command actually generates a usage statement from the base-command which doesn't provide helpful context on viable parameters for that sub-command
```
usage: /usr/bin/cloud-init [-h] [--version] [--file FILES] [--debug] [--force] {init,modules,single,query,features,analyze,devel,collect-logs,clean,status,schema} ...
/usr/bin/cloud-init: error: unrecognized arguments: --invalid
```

## Environment details
- Cloud-init version: 23.4~3g0cb0b80f-0ubuntu1
- Operating System Distribution: ANY
- Cloud provider, platform or installer type: ANY

## cloud-init logs
n/a

**Comment by Dillonsd** [comment]
Hello, I'd love to pick this up. Can I be assigned to this?

**Comment by TheRealFalcon** [comment]
@Dillonsd , yes, feel free to work on this! We usually don't assign issues directly, so your comment here is enough to show that you are working on this.

**Comment by Dillonsd** [comment]
@TheRealFalcon Thanks! PR is up

Also I noticed there was a change made to the CLA signing process. The default markdown comment when you open a PR still asks the contributor to add their username to tools/.github-cla-signers. That might be confusing.
